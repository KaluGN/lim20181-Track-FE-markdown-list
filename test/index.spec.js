const mdlinks = require('../index');

test('mdlinks debería retornar una promesa', () => {
    const options = {
        validate: false,
        stats: false
    }
    return expect(mdlinks("README.md", options)).toBeInstanceOf(Promise);
});

test('Debería retornar [{Total : 6, Unique : 8, Broken : 3}]', () => {
    const options = {
        validate: true,
        stats: true
    }
    return expect(Promise.resolve(mdlinks('test', options))).resolves.toMatchSnapshot();
});

test('Debería retornar [{Total : 6, Unique: 8}]', () => {
    const options = {
        validate: false,
        stats: true
    }
    return expect(Promise.resolve(mdlinks("test", options))).resolves.toMatchSnapshot();
});

test('Debería retornar "Dead" o "alive"', () => {
    const options = {
        validate: true,
        stats: false
    }
    return expect(Promise.resolve(mdlinks("test", options))).resolves.toMatchSnapshot();
});