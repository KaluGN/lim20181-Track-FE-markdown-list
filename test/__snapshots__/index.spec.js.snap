// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debería retornar "Dead" o "alive" 1`] = `
Array [
  Object {
    "href": "https://goo.gl/fbAQLP",
    "status": 200,
    "statusText": "alive",
  },
]
`;

exports[`Debería retornar [{Total : 6, Unique : 8, Broken : 3}] 1`] = `
Array [
  Object {
    "broken": 0,
    "total": 1,
    "unique": 1,
  },
]
`;

exports[`Debería retornar [{Total : 6, Unique: 8}] 1`] = `
Array [
  Object {
    "total": 1,
    "unique": 1,
  },
]
`;
